pydsa/radix_sort.py

```diff
@@ -0,0 +1,61 @@
+"""Radix Sort
+Complexity: Average Case: O(nlog(n)), where log is taken to the
+base for representing numbers. This implementation is for decimal
+system i.e. base is taken to be 10.
+Reference used: http://www.geeksforgeeks.org/radix-sort/
+"""
+
+BASE = 10
+
+
+def radix_sort(a):
+    """
+    Sorts the list 'a' using Radix Sort algorithm.
+
+    >>> from pydsa import radix_sort
+    >>> a = [708, 4567, 3, 45, 911, 123, 57, 37]
+    >>> radix_sort(a)
+    [3, 37, 45, 57, 123, 708, 911, 4567]
+
+    """
+
+    max_element = max(a)
+    n = len(a)
+
+    # Do counting sort for every digit. Note that instead
+    # of passing digit number, exp is passed. exp is BASE^i
+    # where i is current digit number
+    exp = 1
+    while max_element//exp > 0:
+        _counting_sort(a, exp, n)
+        exp *= BASE
+
+    return a
+
+
+def _counting_sort(a, exp, n):
```

Comment:
We have counting sort implemented here: https://github.com/pydsa/pydsa/blob/master/pydsa/counting_sort.py

You can use that, either.